### PR TITLE
[StandAlonePlayer] fix error printing causing a segfault

### PR DESCRIPTION
**Pull request by PoroCYon**
glibc (and possibly others) don't like NULL being passed to `fputs`.

### CODE REVIEW

File: WaveSabreStandAlonePlayer/main.cpp [hunk 1]
```diff
@@ -4,6 +4,15 @@ using namespace WaveSabrePlayerLib;
 
 #include <string.h>
 
+#if defined(_MSC_VER)
+// MSVC WHYYYYYY
+extern "C" FILE* __cdecl __iob_func(void)
```
Comment (yupferris) [inline WaveSabreStandAlonePlayer/main.cpp:9]: What's the issue here? That `fprintf` usage adds a dependency to this function which then isn't resolved at link time? Is there another way around this (eg. passing a proper parameter to `fputs` instead of `NULL`) that avoids this?

File: WaveSabreStandAlonePlayer/main.cpp [hunk 2]
```diff
@@ -118,4 +127,4 @@ int main(int argc, char **argv)
 
 	free(buffer);
 	return 0;
-}
\ No newline at end of file
+}
```
Comment (yupferris) [inline WaveSabreStandAlonePlayer/main.cpp:130]: Unnecessary whitespace change.

**Review by yupferris — CHANGES_REQUESTED** [review]

File: WaveSabreStandAlonePlayer/main.cpp (hunk 1 above)
Comment (PoroCYon) [reply WaveSabreStandAlonePlayer/main.cpp:9]: This isn't for `fprintf`, but for `stderr`: `stderr` gets `#define`d as something like `&__iob_func()[0]`, and the latter isn't included as part of the runtime when compiling with size-optimizations (if I understood the stackoverflow stuff correctly, I don't know anything about how MSVC works, I just look at [the AppVeyor logs](https://ci.appveyor.com/project/yupferris/wavesabre/builds/34930087/job/1d3efhi3xpduh6kj))

The reason I changed it to `fprintf` was that I hardly ever use `fputs`, so I wouldn't know how it handles the lack of a newline at the end (which, right before exiting, might be annoying on many terminal emulators on Linux, at least).

Does Windows have `dprintf`? Because if so, it could probably be changed to use `dprintf(2, "...");`.

File: WaveSabreStandAlonePlayer/main.cpp (hunk 2 above)
Comment (PoroCYon) [reply WaveSabreStandAlonePlayer/main.cpp:130]: Oh, that's vim doing these automatically. I'll change it.

File: WaveSabreStandAlonePlayer/main.cpp (hunk 1 above)
Comment (yupferris) [reply WaveSabreStandAlonePlayer/main.cpp:9]: I took a look at other places where we print stuff built in size-optimized contexts, eg. [PlayerTest](https://github.com/logicomacorp/WaveSabre/blob/master/Tests/PlayerTest/main.cpp#L28), and it seems there we just use `printf` and manually add a newline. I'm not entirely sure how/why we ended up with `fputs` here instead, but it seems `printf` should be a working alternative and avoid extra linker errors etc.

Comment (PoroCYon) [reply WaveSabreStandAlonePlayer/main.cpp:9]: Alright, I'll change it to that, then.

**Review by yupferris — APPROVED** [review]
